### PR TITLE
Add icon policy documentation

### PR DESCRIPTION
**Pull request by Bosch-0**
I created some documentation as to how icons should be prepared, optimized, styled, contributed and attributed within Bitcoin Core. Having guidelines for iconography helps with making applications more consistent and efficient. 

The current icon set does not yet follow this policy but I will be updating them in future PRs. To see the current icon set using this policy see this [figma source file](https://www.figma.com/file/0oqnohjahRtprjRyaetDOL/Design-System?node-id=1881%3A6).

### CODE REVIEW

File: doc/icon_policy.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+## Contributing
+Icons should only be added to the Bitcoin Core repo via a pull request (PR) in two situations:
+
+- An icon used in production is being replaced with a new icon.
+- A feature is being added to the Bitcoin Core GUI that requires the addition of a new icon.
+
+If a new icon is designed for an upcoming or active PR, the designer of the icon should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) to get feedback on the design and to make sure it is consistent with Bitcoin Core's iconography.
+
+If a new feature is being added to the Bitcoin Core GUI that requires a new icon the developer should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) requesting an icon to be designed.
```
Comment (jarolrod) [inline doc/icon_policy.md:49]: NIT reword
```suggestion
If a new feature which requires a new icon is being added to the Bitcoin Core GUI, the developer should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) requesting the new icon be designed. 
```

File: doc/icon_policy.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- An icon used in production is being replaced with a new icon.
+- A feature is being added to the Bitcoin Core GUI that requires the addition of a new icon.
+
+If a new icon is designed for an upcoming or active PR, the designer of the icon should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) to get feedback on the design and to make sure it is consistent with Bitcoin Core's iconography.
+
+If a new feature is being added to the Bitcoin Core GUI that requires a new icon the developer should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) requesting an icon to be designed. 
+
+Icons (both SVG and PNG) should not be added to the Bitcoin Core repo if they are not yet used in production.
+
+When opening a PR that adds an icon source (SVG) files should be added to `src/qt/res/src` at 32x32px and optimized production (PNG) files should be added to `src/qt/res/icons` at 32x32px.
```
Comment (jarolrod) [inline doc/icon_policy.md:53]: NIT reword
Question: both files must be 32x32px, what about the special case where 32x32px is too small?
```suggestion
When opening a PR that intends to add an icon, a SVG source file must be included under `src/qt/res/src`, and a optimized production file (PNG) must be included under `src/qt/res/icons`. Both source and production files should be 32x32px in size.
```

File: doc/icon_policy.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+If a new feature is being added to the Bitcoin Core GUI that requires a new icon the developer should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) requesting an icon to be designed. 
+
+Icons (both SVG and PNG) should not be added to the Bitcoin Core repo if they are not yet used in production.
+
+When opening a PR that adds an icon source (SVG) files should be added to `src/qt/res/src` at 32x32px and optimized production (PNG) files should be added to `src/qt/res/icons` at 32x32px. 
+
+
+
+## Attribution 
+Icon additions should have the appropriate attribution to the author, the licence used for the icon and any comments relevant to the icon included in the
```
Comment (jarolrod) [inline doc/icon_policy.md:58]: NIT reword
```suggestion
Icon additions must include appropriate attribution to the author, license information, and any comments relevant to the icon documented under the
```

**Review by jarolrod — CHANGES_REQUESTED** [review]

File: doc/icon_policy.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+Icons should only be added to the Bitcoin Core repo via a pull request (PR) in two situations:
+
+- An icon used in production is being replaced with a new icon.
+- A feature is being added to the Bitcoin Core GUI that requires the addition of a new icon.
+
+If a new icon is designed for an upcoming or active PR, the designer of the icon should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) to get feedback on the design and to make sure it is consistent with Bitcoin Core's iconography.
+
+If a new feature is being added to the Bitcoin Core GUI that requires a new icon the developer should [open an issue](https://github.com/bitcoin-core/gui/issues/new/choose) requesting an icon to be designed. 
+
+Icons (both SVG and PNG) should not be added to the Bitcoin Core repo if they are not yet used in production.
```
[the remark this answers is not in the document]
Comment (jarolrod) [reply doc/icon_policy.md:51]: While here: this could just say `Icons are not to be added to Bitcoin Core prior to a production use case`

**Comment by jonasschnelli** [comment]
Concept ACK. Great work.
My only question would be, it this should belong to the root `docs/` directory or if we should place it in `src/qt/res/`.

File: doc/icon_policy.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
+# Icon policy
+This document provides guidelines for contributing icons to Bitcoin Core.
+
+
+
+## Preparing Icons
+Both an icon source file, in Scalable Vector Graphics (SVG) format, and an optimised production file, in Portable Network Graphics (PNG) format, are required for
```
[the remark this answers is not in the document]
Comment (Bosch-0) [reply doc/icon_policy.md:7]: My mistake, I write in British English - will make this change to be more consistent with other documentation.

File: doc/icon_policy.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+
+### Icon Style
+Icon style should follow a few basic principles to maintain consistency and clarity:
+
+- Icons should be minimal with little to no fine details. This is to make sure icons look good at any scale.  
+- Icons size should be 32x32px. Different sized icons sizes may be necessary in some circumstances. If 32x32px is too small for the required icon placement, then the icon should be scaled up or down whilst sticking to the 8-point grid system.
+- Shapes should be consistent and avoid organic elements that do not scale well.
+- Colors should be consistent with other icons used in Bitcoin Core. 
+- Icons should should follow an 8-point grid system (see above).
```
[the remark this answers is not in the document]
Comment (Bosch-0) [reply doc/icon_policy.md:31]: Good point, could probably remove this

File: doc/icon_policy.md (hunk 2 above)
Comment (Bosch-0) [reply doc/icon_policy.md:55]: Than the SVG should be used to create a larger production (PNG) file. I'll specify something about this just being the default size - different sizes will have to be tested by those contributing icons to see what size works best.

**Review by Bosch-0 — COMMENTED** [review]
Thanks for the suggestions :)

**Comment by RandyMcMillan** [comment]
I like the idea of putting it in the /res folder.
Maybe even just name it README.md if it lands in the /res folder?
This may be a good place to add logo/branding specification notes in the future.

**Comment by RandyMcMillan** [comment]
This may be helpful in getting this PR to pass.

https://github.com/bitcoin-core/gui/pull/139#issuecomment-739078392

**Comment by Bosch-0** [comment]
>  My only question would be, it this should belong to the root docs/ directory or if we should place it in src/qt/res/.

Yes this is probably a better location for this, I'll move the file and squash the commits

**Comment by jarolrod** [comment]
You should probably squash the commits

**Comment by RandyMcMillan** [comment]
I still think it should be named README.md
There is other documentation we could add to this later. Why not be proactive?

Otherwise ACK

**Comment by Bosch-0** [comment]
Moved https://github.com/bitcoin-core/gui/pull/178

**Pull request closed without merging**